### PR TITLE
return int from all main() functions

### PR DESCRIPTION
**Pull request by DerDakon**

### CODE REVIEW

File: bouncesaying.c [hunk 1]
```diff
@@ -7,14 +7,12 @@
...
-void main(argc,argv)
-int argc;
-char **argv;
+int main(int argc, char **argv)
 {
   int pid;
   int wstat;
 
-  if (!argv[1])
+  if (argc == 1 || !argv[1])
```
Comment (josuah) [inline bouncesaying.c:15]: I did not know there could be any situation `!argv[1] && argc ==1` could happen.

File: forward.c [hunk 2]
```diff
@@ -25,13 +25,12 @@ substdio ssout = SUBSTDIO_FDBUF(mywrite,-1,outbuf,sizeof outbuf);
...
 
-void main(argc,argv)
-int argc;
-char **argv;
+int main(int argc, char **argv)
 {
   char *sender;
   char *dtline;
   char *qqx;
+  int i;
```
Comment (josuah) [inline forward.c:33]: Given `i` is also declared in `for (int i; ...; ...) { ... }`, is it really needed here?
Sorry if I missed it.

File: hostname.c [hunk 3]
```diff
@@ -1,17 +1,16 @@
 #include "substdio.h"
 #include "subfd.h"
 #include "readwrite.h"
-#include "exit.h"
 
 char host[256];
 
-void main()
+int main()
```
Comment (josuah) [inline hostname.c:7]: while at it, do we also convert it to `int main(void)` ?

File: bouncesaying.c (hunk 1 above)
Comment (leahneukirchen) [reply bouncesaying.c:15]: I think just checking argc is good enough?

Comment (DerDakon) [reply bouncesaying.c:15]: It probably is. Let me do some experiments.

File: chkspawn.c [hunk 4]
```diff
@@ -2,7 +2,6 @@
 #include "subfd.h"
 #include "fmt.h"
 #include "select.h"
-#include "exit.h"
```
Comment (schmonz) [inline chkspawn.c:5]: `Makefile` still has `chkspawn.o` depending on `exit.h`

File: dnsfq.c [hunk 5]
```diff
@@ -5,26 +5,25 @@
 #include "dnsdoe.h"
 #include "ip.h"
 #include "ipalloc.h"
-#include "exit.h"
```
Comment (schmonz) [inline dnsfq.c:8]: Makefile dependency thing again

File: dnsip.c [hunk 6]
```diff
@@ -5,7 +5,6 @@
 #include "dnsdoe.h"
 #include "ip.h"
 #include "ipalloc.h"
-#include "exit.h"
```
Comment (schmonz) [inline dnsip.c:8]: You already got the idea last time :-)

File: auto-int.c [hunk 7]
```diff
@@ -13,15 +13,14 @@ char *s;
...
-int argc;
-char **argv;
+int main(int argc, char **argv)
 {
   char *name;
   char *value;
   unsigned long num;
   char strnum[FMT_ULONG];
 
+  if (argc != 3) _exit(100);
```
[the remark this answers is not in the document]
Comment (DerDakon) [reply auto-int.c]: -Wunused-parameter or so

**Pull request merged**